lib/agent.js

```diff
@@ -191,8 +193,8 @@ Agent.prototype.start = function (opts) {
       agent: version,
       ancestors: _ancestors,
       startTrace: stackObj.stack.split(/\n */).slice(1),
-      main: pkg && pkg.main,
-      dependencies: pkg && pkg.dependencies,
+      main: pkg ? pkg.main : '<could not determine>',
+      dependencies: pkg ? pkg.dependencies : '<could not determine>',
       conf: this._conf.toJSON()
```

Comment:
I see when assigning the `main` and `dependencies` properties to the object we're logging that we're using a ternary to provide a default value of `"<could not determine>"` if there was some issue loading `package.json`.  If `package.json` was loaded but `pkg.main` or `pkg.dependencies` is not set then a value of _undefined_ will still be recorded (which seems correct/accurate)